crypto/CMakeLists.txt

```diff
@@ -33,8 +33,16 @@ if(NOT OPENSSL_NO_ASM)
     set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -Wa,--noexecstack")
 
     # Clang's integerated assembler does not support debug symbols.
-    if(NOT CMAKE_ASM_COMPILER_ID MATCHES "Clang")
-      set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -Wa,-g")
+    if(CMAKE_ASM_COMPILER_ID)
```

Comment:
Proposal for simplifying logic. In pseudo-code:
```
if (CMAKE_ASM_COMPILER_ID MATCHES "Clang" || CMAKE_ASM_COMPILER MATCHES "clang")
  message(STATUS "Disabling debug symbols for Clang internal assembler)
else ()
 set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -Wa,-g")
endif()
``

Reply:
This logic also doesn't capture when user specifies an external assembler - but this was not captured previously as well.

Reply:
Thanks. That is much better.